firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -143,9 +143,12 @@ public Thread newThread(Runnable r) {
    * or empty.
    */
   private void preConditionChecks() {
-    Preconditions.checkNotEmpty(getApplicationId());
-    Preconditions.checkNotEmpty(getProjectIdentifier());
-    Preconditions.checkNotEmpty(getApiKey());
+    Preconditions.checkNotEmpty(getApplicationId(), "Firebase App ID should not be empty or null.");
+    Preconditions.checkNotEmpty(
+        getProjectIdentifier(), "Firebase project ID should not be empty or null.");
+    Preconditions.checkNotEmpty(
+        getApiKey(),
+        "Firebase API key used for authenticating requests from your app should not be empty or null.");
```

Comment:
Please set a valid API key. A Firebase API key is required to communicate with Firebase server APIs: It authenticates your project with Google.